CHANGES.md

```diff
@@ -1,3 +1,9 @@
+## 1.16.0
+
+- `write_item` single item write helper for the `write_versioned`
```

Comment:
When would you recommend we use this or not use it? I think I would lean toward using it all the time.

Reply:
so, i've added a bit of documentation to it. but i also had a long conversation with Josh about it, and let me see if I can summarize.

This is super _generic_, which means that it's _not_ going to be the best choice for cases where you want the narrower, more specific behavior offered by the other 3 existing helpers.

As an example - if you actually want to raise an exception when somebody tries to update a job that literally doesn't exist, then you should definitely use `update_existing` - it's the only one that promises it'll treat that situation as an exceptional case.

Similarly, I would advise using create_or_update if you definitely never intend to delete the item - as you can see, returning None will actually cause deletion, and it's actually quite common in our systems to never hard delete items. So don't use one where the type signature would allow you to delete the item if you never intend to be deleting it. 

Overall, I would say this should be used incredibly rarely at XOi. I have introduced it because I do have a specific case where I wanted to be able to fully unit test some existing hard-deletion code. And there would be other circumstances where we might theoretically want to do hard deletions. But in order of more narrow to more general, I'd order these 4 helpers:

1. update_existing
2. update_if_exists
3. create_or_update
4. write_item

And I would suggest we always pick the _narrowest_ one that is applicable to our situation. I think that makes our code _simpler_ even if it's not as _easy_ as just always picking the same helper off the tree.